File: internal/types/journal_rank.go
package types

import "time"

// JournalRankMetadata is the cached UI-facing result of a journal ranking
// lookup. It is stored inside Knowledge.Metadata to avoid a schema migration.
type JournalRankMetadata struct {
	SchemaVersion int                        `json:"schema_version"`
	Publication   string                     `json:"publication"`
	MatchedAt     time.Time                  `json:"matched_at"`
	Source        string                     `json:"source"`
	Found         bool                       `json:"found"`
	Official      map[string]string          `json:"official,omitempty"`
	OfficialAll   map[string]string          `json:"official_all,omitempty"`
	Custom        []JournalRankCustomDataset `json:"custom,omitempty"`
}

type JournalRankCustomDataset struct {
	Label string `json:"label"`
	Level string `json:"level"`
}

File: frontend/src/types/journalRank.ts
export interface JournalRankCustomDataset {
  label: string
  level: string
}

export interface JournalRankMetadata {
  schema_version: number
  publication: string
  matched_at: string
  source: string
  found: boolean
  official?: Record<string, string>
  official_all?: Record<string, string>
  custom?: JournalRankCustomDataset[]
}

export interface KnowledgeJournalMetadata {
  journal_rank?: JournalRankMetadata
  [key: string]: unknown
}
